go/weed/weed_server/filer_server_handlers.go

```diff
@@ -134,14 +161,63 @@ func (fs *FilerServer) PostHandler(w http.ResponseWriter, r *http.Request) {
 	if collection == "" {
 		collection = fs.collection
 	}
-	assignResult, ae := operation.Assign(fs.master, 1, replication, collection, query.Get("ttl"))
-	if ae != nil {
-		glog.V(0).Infoln("failing to assign a file id", ae.Error())
-		writeJsonError(w, r, http.StatusInternalServerError, ae)
+
+	buf, _ := ioutil.ReadAll(r.Body)
+	r.Body = analogueReader{bytes.NewBuffer(buf)}
+	fileName, _, _, _, _, _, _, pe := storage.ParseUpload(r)
+	if pe != nil {
+		glog.V(0).Infoln("failing to parse post body", pe.Error())
+		writeJsonError(w, r, http.StatusInternalServerError, pe)
+		return
+	}
+	//reconstruct http request body for following new request to volume server
+	r.Body = analogueReader{bytes.NewBuffer(buf)}
```

Comment:
Chris, so far I could not think of a better way to avoid this additional http body copy,  could you give me some suggestion?
if we restrict above code only be implemented when end user trigger an update operation(HTTP PUT instead of HTTP POST), it will be easier and follow the restful API's rules.
Please feel free to give me your opinion and point out my mistake~ Thanks~